runtime/compiler/env/ClassLoaderTable.cpp

```diff
@@ -182,28 +203,29 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
    if (info)
       return;
 
-#if defined(J9VM_OPT_JITSERVER)
-   bool useAOTCache = _persistentMemory->getPersistentInfo()->getJITServerUseAOTCache();
-   const J9UTF8 *nameStr = J9ROMCLASS_CLASSNAME(((J9Class *)clazz)->romClass);
-   const uint8_t *name = J9UTF8_DATA(nameStr);
-   uint16_t nameLength = J9UTF8_LENGTH(nameStr);
-#endif /* defined(J9VM_OPT_JITSERVER) */
 
-   void *chain = _sharedCache->rememberClass(clazz);
-   if (!chain)
-      {
-#if defined(J9VM_OPT_JITSERVER)
-      if (useAOTCache && TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
-            "ERROR: Failed to get class chain for %.*s loaded by %p",
-            nameLength, (const char *)name, loader
-         );
-#endif /* defined(J9VM_OPT_JITSERVER) */
+   void *chain = _sharedCache ? _sharedCache->rememberClass(clazz) : NULL;
+   // If we are using the JITServer AOT cache, we need to remember the name of clazz with or without chain.
+   // If we aren't using it, there is no point in continuing.
+   if (!chain && !useAOTCache)
```

Comment:
Ok, just a note than in the future we may not have an AOT cache or a SCC and still want to do this.

Reply:
Yes, there's not much point to doing this without the upcoming JITServer AOT cache compilation changes.